apps/client/src/register/components/registerForm.tsx

```diff
@@ -0,0 +1,154 @@
+import { Button, TextField, Typography, Box, Container } from '@mui/material';
+import { z } from 'zod';
+import React, { useState } from 'react';
+import axios from 'axios';
+import Alert from '@mui/material/Alert';
+interface RegisterFormProps {
+  email: string;
+  password: string;
+  confirmPassword: string;
+  firstName: string;
+  lastName: string;
+}
+
+const initialFormValues: RegisterFormProps = {
+  email: '',
+  password: '',
+  confirmPassword: '',
+  firstName: '',
+  lastName: '',
+};
+
+const registerFormSchema = z
+  .object({
+    email: z.string().email().max(255),
+    password: z.string().min(6),
+    confirmPassword: z.string(),
+    firstName: z.string().max(30),
+    lastName: z.string().max(30),
+  })
+  .refine((data) => data.password === data.confirmPassword, {
+    message: 'Passwords do not match',
+    path: ['confirmPassword'],
+  });
+
+const RegisterForm: React.FC = () => {
+  const [formValues, setFormValues] =
+    useState<RegisterFormProps>(initialFormValues);
+  const [errors, setErrors] = useState<string[]>([]);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormValues((prevValues) => ({
+      ...prevValues,
+      [name]: value,
+    }));
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    try {
+      const { confirmPassword, ...validatedData } =
+        registerFormSchema.parse(formValues);
+
+      const response = await axios.post(
+        'http://localhost:3000/api/v1/register',
+        validatedData
+      );
+      console.log(response);
+    } catch (error) {
+      if (error instanceof z.ZodError) {
+        const validationErrors = error.errors.map((err) => err.message);
+
+        setErrors(validationErrors);
+      } else {
+        console.error(error);
+      }
+    }
+  };
```

Comment:
Room for improvement, with service layer.
Create a new state for managing the `errors`